src/comments/index.ts

```diff
@@ -0,0 +1,190 @@
+import { sha256 } from 'js-sha256';
+import showQuickEditor from '../shared/show-quick-editor';
+import { Hovercards } from '@gravatar-com/hovercards';
+import './style.scss';
+
+const BASE_API_URL = 'https://api.gravatar.com/v3/profiles';
+const GRAVATAR_CONTAINER = '.gravatar-enhanced-profile';
+const COMMENT_EMAIL_FIELD = '#email';
+const INPUT_TIMEOUT = 1000;
+const DEBOUNCE_TIMEOUT = 250;
+
+const hovercards = new Hovercards();
+
+function isEmail( email ) {
+	const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+	return emailRegex.test( email );
+}
+
+async function fetchUserProfile( email ) {
+	const hash = sha256( email.trim().toLowerCase() );
+
+	try {
+		// Get profile data
+		const response = await fetch( `${ BASE_API_URL }/${ hash }?source=hovercard` );
+		if ( ! response.ok ) {
+			return null;
+		}
+
+		return await response.json();
+	} catch ( error ) {
+		// eslint-disable-next-line no-console
+		console.error( error );
+	}
+
+	return null;
+}
+
+function suggestProfile( profile ) {
+	const author = document.getElementById( 'author' ) as HTMLInputElement;
+	const url = document.getElementById( 'url' ) as HTMLInputElement;
+	const firstVerified = profile.verified_accounts.length > 0 ? profile.verified_accounts[ 0 ] : null;
+
+	if ( author && author.value === '' ) {
+		author.value = profile.display_name;
+	}
+
+	if ( url && url.value === '' && firstVerified ) {
+		url.value = firstVerified.url;
+	}
+}
+
+function hideProfile() {
+	const emailContainer = document.querySelector( '.comment-form-email' ) as HTMLInputElement;
+	const emailField = document.querySelector( COMMENT_EMAIL_FIELD ) as HTMLInputElement;
+
+	if ( ! emailField || ! emailContainer ) {
+		return;
+	}
+
+	emailContainer.classList.remove( 'gravatar-enhanced-comments' );
+	emailField.style.paddingLeft = '';
+}
+
+function adjustGravatarPosition() {
+	const gravatarProfile = document.querySelector( GRAVATAR_CONTAINER ) as HTMLElement;
```

Comment:
Nit: The class has been added to the `<span/>`:

```suggestion
	const gravatarProfile = document.querySelector( GRAVATAR_CONTAINER ) as HTMLSpanElement;
```